gunicorn/arbiter.py

```diff
@@ -365,9 +378,8 @@ def stop(self, graceful=True):
         killed gracefully  (ie. trying to wait for the current connection)
         """
 
-        if self.reexec_pid == 0 and self.master_pid == 0:
-            for l in self.LISTENERS:
-                l.close()
+        unlink = self.reexec_pid == self.master_pid and not self.systemd
```

Comment:
I don't think there's any way they can be equal unless they are zero, but I agree it would be easier to read and safer to be explicit. Fixing.